File: 1/main.cpp
#include <stdlib.h>
#include <stdio.h>
#include <chrono>

typedef int ** matrix;

matrix create_matrix(int n)
{
    int ** matrix = new int*[n];
    for (int i = 0; i < n; ++i)
    {
        matrix[i] = new int[n];
    }
    return matrix;
}

matrix mult_matrix(matrix a, matrix b, int n)
{
    matrix c = create_matrix(n);
    for (int i = 0; i < n; ++i)
    {
        for (int j = 0; j < n; ++j)
        {
            for (int k = 0; k < n; ++k)
            {
                c[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    return c;
}

int main(int argc, char ** argv)
{
    if (argc != 2)
    {
        printf("Usage: %s <n>\n", argv[0]);
        return 1;
    }

    int n = atoi(argv[1]);

    matrix a = create_matrix(n);
    matrix b = create_matrix(n);
    
    auto begin = std::chrono::system_clock::now();
    mult_matrix(a, b, n);
    auto end = std::chrono::system_clock::now();

    printf("%f\n", std::chrono::duration<double>(end - begin).count());
    return 0; 
}

File: 1/main.py
import sys
import time

def create_matrix(n):
    matrix = [];
    for i in range(n):
        matrix.append([0] * n)
    return matrix

def mult_matrix(a, b, n):
    c = create_matrix(n)
    for i in range(n):
        for j in range(n):
            for k in range(n):
                c[i][j] += a[i][k] * b[k][j]
    return c

if len(sys.argv) != 2:
    print("Usage: python {} <n>".format(sys.argv[0]))
    sys.exit()

n = int(sys.argv[1])

a = create_matrix(n)
b = create_matrix(n)

start_time = time.time()
c = mult_matrix(a, b, n)
print(time.time() - start_time)
    
